content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,381 @@
+
+```shell
+kubectl create configmap fruits --from-literal=fruits=apples
+```
+
+Below is an example of a Deployment manifest with an environment variable configured via the ConfigMap `fruits`.
+{{% code_sample file="deployments/deployment-with-configmap-as-envvar.yaml" %}}
+
+Create the deployment.
+```shell
+kubectl apply -f https://k8s.io/examples/deployments/deployment-with-configmap-as-envvar.yaml
+```
+
+Check the deployment.
+```shell
+kubectl get deployment configmap-env-var
+```
+
+You should see an output similar to:
+```
+NAME                READY   UP-TO-DATE   AVAILABLE   AGE
+configmap-env-var   3/3     3            3           7s
+```
+
+Check the pods.
+```shell
+kubectl get pods --selector=app=configmap-env-var
+```
+
+You should see an output similar to:
+```
+NAME                                 READY   STATUS    RESTARTS   AGE
+configmap-env-var-59cfc64f7d-74d7z   1/1     Running   0          46s
+configmap-env-var-59cfc64f7d-c4wmj   1/1     Running   0          46s
+configmap-env-var-59cfc64f7d-dpr98   1/1     Running   0          46s
+```
+
+The key-value pair in the ConfigMap is configured as an environment variable in the container of the Pod.
+Check this by viewing the logs of the Pod.
+```shell
+kubectl logs deployment/configmap-env-var
+```
+
+You should see an output similar to the below:
+```
+Found 3 pods, using pod/configmap-env-var-7c994f7769-l74nq
+Thu Jan  4 16:07:06 UTC 2024 The basket is full of apples
+Thu Jan  4 16:07:16 UTC 2024 The basket is full of apples
+Thu Jan  4 16:07:26 UTC 2024 The basket is full of apples
+```
+
+Edit the ConfigMap.
+```shell
+kubectl edit configmap fruits
+```
+
+In the editor that appears, change the value of key `fruits` from `apples` to `mangoes`. Save your changes.
+The kubectl tool updates the ConfigMap accordingly (if you see an error, try again).
+
+Here's an example of how that manifest could look after you edit it:
+```yaml
+apiVersion: v1
+data:
+  fruits: mangoes
+kind: ConfigMap
+# You can leave the existing metadata as they are.
+# The values you'll see won't exactly match these.
+metadata:
+  creationTimestamp: "2024-01-04T16:04:19Z"
+  name: fruits
+  namespace: default
+  resourceVersion: "1749472"
+
+```
+
+You should see the following output:
+```
+configmap/fruits edited
+```
+
+Tail the logs of the Deployment and observe the output for few seconds:
+```shell
+kubectl logs deployment/configmap-env-var
+```
+
+You will notice that the output remains unchanged:
+```
+Thu Jan  4 16:12:56 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:06 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:16 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:26 UTC 2024 The basket is full of apples
+
+```
+
+{{< note >}}
+Although the value of the key inside the ConfigMap has changed, the environment variable in the Pod still possesses the earlier value. This is because environment variables in a Pod are not updated until the Pod is rolled-over.
```

Comment:
```suggestion
Although the value of the key inside the ConfigMap has changed, the environment variable in the Pod still
shows the earlier value. This is because environment variables for a process running inside a Pod are
**not** updated when the source data changes; if you wanted to force an update, you would need to
have Kubernetes replace your existing Pods. The new Pods would then run with the updated information.
```

Reply:
Ack, will be included in the next commit.